### PR TITLE
Jazzy updates

### LINKED ISSUE #783: Update public docs for Jazzy's Swift formatting
We use `jazzy` for our docs, but the current formatting is still a copy/paste from the obj-c version, and Jazzy has a different format for Swift. 

Jazzy's docs are here: https://github.com/realm/jazzy#supported-documentation-keywords
We should also try to make it compatible with DocC, since Jazzy added support for that syntax as well, and it'll be useful for folks looking up docs within Xcode. 

Here's an example file using the Objective-C format for jazzy docs: https://github.com/RevenueCat/purchases-ios/blob/swift_migration/Purchases/Public/PurchasesDelegate.swift#L29

### PR DESCRIPTION
**Pull request by taquitos**
Resolves #783

### CODE REVIEW

File: Purchases/Public/Error Handling/ErrorUtils.swift [hunk 1]
```diff
@@ -48,14 +47,14 @@ import StoreKit
...
+     * `RCUnderlyingErrorKey` in the `NSError.userInfo` dictionary. The backend error code will be mapped using
+     * `BackendErrorCode.toPurchasesErrorCode()`.
      *
-     * - Parameter: backendCode The numerical value of the error.
-     * - Parameter: backendMessage The message of the errror contained under the [NSUnderlyingErrorKey] key in the
+     * - Parameter backendCode: The numerical value of the error.
+     * - Parameter backendMessage: The message of the errror contained under the `NSUnderlyingErrorKey` key in the
      * UserInfo dictionary.
-     * - Parameter: finishable Will be added to the UserInfo dictionary under the [RCFinishableKey] to indicate if the
+     * - Parameter finishable: Will be added to the UserInfo dictionary under the `RCFinishableKey` to indicate if the
```
Comment (aboedo) [inline Purchases/Public/Error Handling/ErrorUtils.swift:57]: this is now `finishableKey`, right?

File: Purchases/Public/Package.swift [hunk 2]
```diff
@@ -96,16 +108,16 @@ private extension PackageType {
...
      */
     static func string(from packageType: PackageType) -> String? {
         return packageType.description
     }
 
     /**
-     * - Parameter string: A string that maps to a enumeration value of type PackageType
-     * - Returns: a PackageType for the given string.
+     * - Parameter string: A string that maps to a enumeration value of type `PackageType`
+     * - Returns: a `PackageType` for the given string.
```
Comment (aboedo) [inline Purchases/Public/Package.swift:120]: I think we should try to do double-backticks for all type references, that way [DocC will auto-link them to the respective types](https://developer.apple.com/documentation/xcode/formatting-your-documentation-content#Link-to-Symbols-and-Other-Content). This is [compatible with Jazzy too](https://github.com/realm/jazzy#supported-documentation-keywords)

Comment (aboedo) [reply Purchases/Public/Package.swift:120]: We can do it in a follow-up pass too

File: Purchases/Public/Purchases.swift [hunk 3]
```diff
@@ -956,12 +957,12 @@ public extension Purchases {
      *
      * - Note: Subscription groups are automatically collected for determining eligibility. If RevenueCat can't
      * definitively compute the eligibilty, most likely because of missing group information, it will return
-     * `RCIntroEligibilityStatusUnknown`. The best course of action on unknown status is to display the non-intro
+     * `IntroEligibilityStatusUnknown`. The best course of action on unknown status is to display the non-intro
```
Comment (aboedo) [inline Purchases/Public/Purchases.swift:960]: `IntroEligibilityStatus.unknown`

**Review by aboedo — COMMENTED** [review]
Looks great! Left a few minor comments. It'd be great to get the type links working with DocC too, but it can be done as a separate thing since this PR already gets us in a much better place.

File: Purchases/Public/Error Handling/ErrorUtils.swift (hunk 1 above)
Comment (taquitos) [reply Purchases/Public/Error Handling/ErrorUtils.swift:57]: It's finishableKey on the error, but the public API for creating the error hasn't changed.

File: Purchases/Public/Package.swift (hunk 2 above)
Comment (taquitos) [reply Purchases/Public/Package.swift]: Done!

File: Purchases/Public/Purchases.swift (hunk 3 above)
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: All the things like this should be fixed now.

File: Purchases/Public/Error Handling/ErrorDetails.swift [hunk 4]
```diff
@@ -17,6 +16,9 @@ import Foundation
 
 @objc(RCErrorDetails) public class ErrorDetails: NSObject {
 
+    /**
+     * These are tacos
+     */
```
Comment (aboedo) [inline Purchases/Public/Error Handling/ErrorDetails.swift:21]: 🌮🌮🌮

File: Purchases/Public/Offering.swift [hunk 5]
```diff
@@ -27,7 +27,7 @@ import Foundation
     @objc public let serverDescription: String
 
     /**
-     Array of `RCPackage` objects available for purchase.
+     Array of `Package` objects available for purchase.
```
Comment (aboedo) [inline Purchases/Public/Offering.swift:30]: double backtick

File: Purchases/Public/Purchases.swift [hunk 6]
```diff
@@ -797,7 +798,7 @@ public extension Purchases {
      * application agreements.
      * If you're having trouble see: https://www.revenuecat.com/2018/10/11/configuring-in-app-products-is-hard
      *
-     * - Parameter identifiers: A set of product identifiers for in app purchases setup via iTunesConnect.
+     * - Parameter identifiers: A set of product identifiers for in app purchases setup via AppStoreConnect.
```
Comment (aboedo) [inline Purchases/Public/Purchases.swift:801]: RIP iTunesConnect. You will not be missed.

Comment (aboedo) [reply Purchases/Public/Purchases.swift:801]: wonder if we should link to the ASC website.

**Review by aboedo — APPROVED** [review]
looks awesome! We probably wanna update the 🌮 doc, but other than that it's ready to go! Long live DocC!

File: Purchases/Public/Error Handling/ErrorDetails.swift (hunk 4 above)
Comment (taquitos) [reply Purchases/Public/Error Handling/ErrorDetails.swift:21]: LOL whoops.

File: Purchases/Public/Purchases.swift (hunk 6 above)
Comment (taquitos) [reply Purchases/Public/Purchases.swift]: ✅ 
Yeah, I think it's super easy to look up, but I don't have a problem adding it as well, so I did that

File: Purchases/Public/Offering.swift (hunk 5 above)
Comment (taquitos) [reply Purchases/Public/Offering.swift]: ✅ , also added it for a few other places where it was `package`

**Pull request merged**